src/IO/WriteHelpers.h

```diff
@@ -821,7 +821,23 @@ inline void writeDateTime64FractionalText(typename DecimalType::NativeType fract
     for (Int32 pos = scale - 1; pos >= 0 && fractional; --pos, fractional /= DateTime64(10))
         data[pos] += fractional % DateTime64(10);
 
-    writeString(&data[0], static_cast<size_t>(scale), buf);
+    if constexpr (trim_suffix_zeros)
```

Comment:
The variable name should also be adjusted.

Reply:
thanks, I will update later.